### PR TITLE
ceph: ignore monitoring folder from yamllint

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit add monitoring folder to yamllint
ignore list.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@anmolsachan ^^

**Comment by anmolsachan** [comment]
> @anmolsachan ^^

@subhamkrai Thanks. Looks good. 

@umangachapagain Please review.

**Review by leseb — CHANGES_REQUESTED** [review]
Typo in commit message.

**Comment by travisn** [comment]
@anmolsachan For future reference, do we have documented steps for generating these yamls? It would help avoid the mistake of touching them with the editor that re-formats the yaml spacing again. Perhaps you could add a readme.md to this folder? Thanks

**Comment by anmolsachan** [comment]
> @anmolsachan For future reference, do we have documented steps for generating these yamls? It would help avoid the mistake of touching them with the editor that re-formats the yaml spacing again. Perhaps you could add a readme.md to this folder? Thanks

Currently, we generate using the ceph-mixins and manually replace the metadata. Perhaps its time we should have the mixin mechanism in rook than ceph. I will discuss this with @umangachapagain and will update you. This manual process has been bugging us for a while.

**Comment by umangachapagain** [comment]
@subhamkrai The title says you are ignoring monitoring from yamllint but the changes look like you are actually doing yamllint. Can you help me understand?
The change looks good though and shouldn't cause any trouble.

**Comment by subhamkrai** [comment]
> @subhamkrai The title says you are ignoring monitoring from yamllint but the changes look like you are actually doing yamllint. Can you help me understand?
> The change looks good though and shouldn't cause any trouble.

If you see changes, you will notice space is removed and `' '` instead of `" "` in this PR which yamllint introduced earlier in my other PR

**Review by umangachapagain — APPROVED** [review]

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 1]
```diff
@@ -8,329 +8,296 @@ metadata:
...
-          for: 0s
-          labels:
-            severity: critical
+  - name: ceph.rules
+    rules:
+    - expr: |
+        kube_node_status_condition{condition="Ready",job="kube-state-metrics",status="true"} * on (node) group_right() max(label_replace(ceph_disk_occupation{job="rook-ceph-mgr"},"node","$1","exported_instance","(.*)")) by (node)
+      record: cluster:ceph_node_down:join_kube
+    - expr: |
+        avg(max by(instance) (label_replace(label_replace(ceph_disk_occupation{job="rook-ceph-mgr"}, "instance", "$1", "exported_instance", "(.*)"), "device", "$1", "device", "/dev/(.*)") * on(instance, device) group_right() (irate(node_disk_read_time_seconds_total[1m]) + irate(node_disk_write_time_seconds_total[1m]) / (clamp_min(irate(node_disk_reads_completed_total[1m]), 1) + irate(node_disk_writes_completed_total[1m])))))
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:17]: Looks like this is rebased on an older commit. It's doing the change here, which should not be.

**Review by anmolsachan — CHANGES_REQUESTED** [review]

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 2]
```diff
@@ -8,329 +8,296 @@ metadata:
...
+          now. Free up some space or expand the storage cluster immediately.
+        message: Storage cluster is read-only now and needs immediate data deletion
+          or cluster expansion.
+        severity_level: error
+        storage_type: ceph
+      expr: |
+        ceph_cluster_total_used_raw_bytes / ceph_cluster_total_bytes >= 0.85
+      for: 0s
+      labels:
+        severity: critical
```
Comment (anmolsachan) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:303]: Also, this file is missing the alerts which I recently added, CephOSDFlapping.

Both files should be the same except for the lint formatting

**Review by anmolsachan — CHANGES_REQUESTED** [review]

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml (hunk 1 above)
Comment (subhamkrai) [reply cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml]: reverted.

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml (hunk 2 above)
Comment (subhamkrai) [reply cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:316]: added.

Comment (anmolsachan) [reply cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:316]: Thanks.

**Review by anmolsachan — APPROVED** [review]

**Comment by anmolsachan** [comment]
@leseb Please review.

**Review by leseb — COMMENTED** [review]
To make sure I understand, we don't only ignore but we also lint right? If so, please split using 2 commits. Thanks.

**Comment by anmolsachan** [comment]
> To make sure I understand, we don't only ignore but we also lint right? If so, please split using 2 commits. Thanks.

@leseb This PR reverses the lint done in https://github.com/rook/rook/pull/6790 and puts the files under the monitoring folder so that in the future they are not linted.

**Review by leseb — APPROVED** [review]

**Pull request merged**